### ISSUE TITLE
Possible Permission Error with Bundle Install in Windows Shells

### ISSUE THREAD

**Issue by aedryan** [bundler/bundler#6758]
On Windows here, updated to 1.17.0 this morning and found that the step that runs `bundle install` when creating a vagrant vm when using git bash, powershell, and cmd results in what seems to be a permissions error.  I have made sure to start the shells as an administrator.  I've posted the environment output from git bash but I've also tested this in powershell and cmd.  Specifying 1.16.6 installs just fine. 
 It's as if it's prompting me for sudo but can't because that's not a thing in these shells.

## Environment
```
Bundler       1.17.0
  Platforms   ruby, x64-mingw32
Ruby          2.3.3p222 (2016-11-21 revision 56859) [x64-mingw32]
  Full Path   C:/Ruby23-x64/bin/ruby.exe
  Config Dir  C:/ProgramData
RubyGems      2.7.7
  Gem Home    C:/Ruby23-x64/lib/ruby/gems/2.3.0
  Gem Path    C:/Users/acox/.gem/ruby/2.3.0;C:/Ruby23-x64/lib/ruby/gems/2.3.0
  User Path   C:/Users/acox/.gem/ruby/2.3.0
  Bin Dir     C:/Ruby23-x64/bin
Tools
  Git         2.17.1.windows.2
  RVM         not installed
  rbenv       not installed
  chruby      not installed
```

## Bundler Build Metadata

```
Built At          2018-10-25
Git SHA           1fe7e93e9
Released Version  true
```

## BackTrace
```
/var/lib/gems/2.5.0/gems/bundler-1.17.0/lib/bundler.rb:374:in `sort'
    default: :
    default: comparison of Pathname with String failed
    default:  (
    default: ArgumentError
    default: )
    default:    from /var/lib/gems/2.5.0/gems/bundler-1.17.0/lib/bundler.rb:374:in `requires_sudo?'
    default:    from /var/lib/gems/2.5.0/gems/bundler-1.17.0/lib/bundler/source/git.rb:212:in `cache_path'
    default:    from /var/lib/gems/2.5.0/gems/bundler-1.17.0/lib/bundler/source/git.rb:299:in `git_proxy'
    default:    from /var/lib/gems/2.5.0/gems/bundler-1.17.0/lib/bundler/source/git.rb:225:in `revision'
    default:    from /var/lib/gems/2.5.0/gems/bundler-1.17.0/lib/bundler/source/git.rb:76:in `to_s'
    default:    from /var/lib/gems/2.5.0/gems/bundler-1.17.0/lib/bundler/source_list.rb:85:in `each'
    default:    from /var/lib/gems/2.5.0/gems/bundler-1.17.0/lib/bundler/source_list.rb:85:in `sort_by'
    default:    from /var/lib/gems/2.5.0/gems/bundler-1.17.0/lib/bundler/source_list.rb:85:in `lock_sources'
    default:    from /var/lib/gems/2.5.0/gems/bundler-1.17.0/lib/bundler/source_list.rb:104:in `replace_sources!'
    default:    from /var/lib/gems/2.5.0/gems/bundler-1.17.0/lib/bundler/definition.rb:674:in `converge_sources'
    default:    from /var/lib/gems/2.5.0/gems/bundler-1.17.0/lib/bundler/definition.rb:120:in `initialize'
    default:    from /var/lib/gems/2.5.0/gems/bundler-1.17.0/lib/bundler/dsl.rb:234:in `new'
    default:    from /var/lib/gems/2.5.0/gems/bundler-1.17.0/lib/bundler/dsl.rb:234:in `to_definition'
    default:    from /var/lib/gems/2.5.0/gems/bundler-1.17.0/lib/bundler/dsl.rb:13:in `evaluate'
    default:    from /var/lib/gems/2.5.0/gems/bundler-1.17.0/lib/bundler/definition.rb:34:in `build'
    default:    from /var/lib/gems/2.5.0/gems/bundler-1.17.0/lib/bundler.rb:135:in `definition'
    default:    from /var/lib/gems/2.5.0/gems/bundler-1.17.0/lib/bundler/cli/install.rb:62:in `run'
    default:    from /var/lib/gems/2.5.0/gems/bundler-1.17.0/lib/bundler/cli.rb:235:in `block in install'
    default:    from /var/lib/gems/2.5.0/gems/bundler-1.17.0/lib/bundler/settings.rb:143:in `temporary'
    default:    from /var/lib/gems/2.5.0/gems/bundler-1.17.0/lib/bundler/cli.rb:234:in `install'
    default:    from /var/lib/gems/2.5.0/gems/bundler-1.17.0/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
    default:    from /var/lib/gems/2.5.0/gems/bundler-1.17.0/lib/bundler/vendor/thor/lib/thor/invocation.rb:126:in `invoke_command'
    default:    from /var/lib/gems/2.5.0/gems/bundler-1.17.0/lib/bundler/vendor/thor/lib/thor.rb:387:in `dispatch'
    default:    from /var/lib/gems/2.5.0/gems/bundler-1.17.0/lib/bundler/cli.rb:27:in `dispatch'
    default:    from /var/lib/gems/2.5.0/gems/bundler-1.17.0/lib/bundler/vendor/thor/lib/thor/base.rb:466:in `start'
    default:    from /var/lib/gems/2.5.0/gems/bundler-1.17.0/lib/bundler/cli.rb:18:in `start'
    default:    from /var/lib/gems/2.5.0/gems/bundler-1.17.0/exe/bundle:30:in `block in <top (required)>'
    default:    from /var/lib/gems/2.5.0/gems/bundler-1.17.0/lib/bundler/friendly_errors.rb:124:in `with_friendly_errors'
    default:    from /var/lib/gems/2.5.0/gems/bundler-1.17.0/exe/bundle:22:in `<top (required)>'
    default:    from /usr/local/bin/bundle:23:in `load'
    default:    from /usr/local/bin/bundle:23:in `<main>'
```

**Comment by colby-swandale** [comment]
This has been fixed in #6761

**Comment by colby-swandale** [comment]
If you update to bundler 1.17.1, your problem should be resolved.
